AlphaWallet/Tokens/ViewControllers/TokensViewController.swift

```diff
@@ -336,7 +314,7 @@ extension TokensViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
 
-        didSelectToken(indexPath: indexPath)
+        selection.send(.cell(indexPath: indexPath))
```

Comment:
Not sure about this one. Changing delegate functions to using `Publisher` reduces number of lines of code, but seems to make it much harder to find callers and maintain them. It's a little like firing NSNotification(s) in Obj-C instead of using delegate function to achieve 1:N handlers and everyone being able to handle it.

They are OK if we attach the sink directly to a button (or any UIControl/source). e.g.

```
button
    .sink {
        foo()
    }
```

but once we want another object to handle the event, it isn't as clear compared to using delegates:

```
let pub = button.somePublisher()
C().something(pub: pub)

class C {
    func something(pub) {
        pub
            .sink {
                foo()
            }
    }
}
```

Or do you think publishers are better here? If so, why?

Reply:
`selection` is applied to hide more logic to view model, since we have different datasources ` i mean different arrays of data`, we want to keep more logic inside view model. selection helps to keep it separatelly.

Reply:
Ok. Let's try it then.